src/ts/fileDescriptorTSServices.ts

```diff
@@ -71,25 +71,27 @@ export function printFileDescriptorTSServices(fileDescriptor: FileDescriptorProt
   printer.printEmptyLn();
 
   fileDescriptor.getServiceList().forEach(service => {
-    printer.printLn(`export class ${service.getName()} {`);
-    printer.printIndentedLn(`static serviceName: string;`);
-    printer.printLn(`}`);
-
-    printer.printLn(`export namespace ${service.getName()} {`);
-    const methodPrinter = new Printer(1);
     service.getMethodList().forEach(method => {
       const requestMessageTypeName = getFieldType(MESSAGE_TYPE, method.getInputType().slice(1), "", exportMap);
       const responseMessageTypeName = getFieldType(MESSAGE_TYPE, method.getOutputType().slice(1), "", exportMap);
-      methodPrinter.printLn(`export class ${method.getName()} {`);
-      methodPrinter.printIndentedLn(`static readonly methodName: string;`);
-      methodPrinter.printIndentedLn(`static readonly service: typeof ${service.getName()};`);
-      methodPrinter.printIndentedLn(`static readonly requestStream: ${method.getClientStreaming()};`);
-      methodPrinter.printIndentedLn(`static readonly responseStream: ${method.getServerStreaming()};`);
-      methodPrinter.printIndentedLn(`static readonly requestType: typeof ${requestMessageTypeName};`);
-      methodPrinter.printIndentedLn(`static readonly responseType: typeof ${responseMessageTypeName};`);
-      methodPrinter.printLn(`}`);
+
+      printer.printLn(`type ${service.getName()}${method.getName()} = {`);
```

Comment:
We now export a `type` for each Service Method in the TypeScript Definition, these types are used to provide the type hint for each exported method on the Service Class (printed below)